gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsTimeStatistic.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2021 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.google.cloud.hadoop.fs.gcs;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Maps;
+import java.util.EnumSet;
+import org.apache.hadoop.classification.InterfaceStability;
+
+/** Time Statistics capturing total time taken by a function in micro seconds */
+@InterfaceStability.Unstable
+public enum GhfsTimeStatistic {
```

Comment:
any plans to capture read/write time?

Reply:
I believe that is captures in hflush, hsync and close statistics from GoogleHadoopOutputStream. If not, can you point to specific functions ?